lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -968,6 +973,11 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgExpression(const DIExpression *Expr) {
     assert(OpCountMap.find(OC) != OpCountMap.end() &&
            "unhandled opcode found in DIExpression");
     unsigned OpCount = OpCountMap[OC];
+    if (isUnsupportedDIE(OC)) {
+      assert(OpCount >= 1);
+      I += OpCount - 1;
+      continue;
```

Comment:
I think, if we just skip an operation we get a wrong expression, don't we?
If we want to avoid a crash, we might:
 * Skip the intrinsic(s)(e.g. `llvm.dbg.value`) which refers to the expression with unsupported operation.
 * Lower the expression so it contains only supported operations, like it is done in [this test](https://github.com/llvm/llvm-project/blob/7b64765cd19a60d6d56106769d06cdff5b660479/llvm/test/DebugInfo/X86/convert-debugloc.ll#L37)

Reply:
> Lower the expression so it contains only supported operations, like it is done in this test

That sounds good, but I cannot find even basic operations like mul or shift in the list of supported [operations](https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.DebugInfo.100.html#_debug_operations_a_id_operation_a). Am I missing something or none of them is supported?
To lower DW_OP_convert we need DW_OP_dup, shr, lit0, not, mul, or.

Reply:
> > Lower the expression so it contains only supported operations, like it is done in this test
> 
> That sounds good, but I cannot find even basic operations like mul or shift in the list of supported [operations](https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.DebugInfo.100.html#_debug_operations_a_id_operation_a). Am I missing something or none of them is supported?
> To lower DW_OP_convert we need DW_OP_dup, shr, lit0, not, mul, or.

You are right, they are not supported in SPIR-V. It seems this option doesn't work in our case.